app/javascript/components/DecisionTree.jsx

```diff
@@ -0,0 +1,290 @@
+import React, { Component } from 'react'
+
+import Title from '../ui/Title'
+import Headline from '../ui/Headline'
+import Field from '../ui/Field'
+import uuidv4 from 'uuid/v4'
+
+import {
+  Button,
+  Paper,
+  TextField
+} from 'react-md'
+
+import RelevanceField from './RelevanceField'
+
+export default class DecisionTree extends Component {
+
+  render() {
+    const { skill, actions } = this.props
+    const { name, config } = skill
+
+    const updateConfig = (key) => {
+      return (value) => {
+        actions.updateSkill({
+          ...skill,
+          config: {
+            ...skill.config,
+            [key]: value
+          }
+        })
+      }
+    }
+
+    return (
+      <div>
+        <Title>Decision tree</Title>
+        <Headline>
+          This skill will guide the user for diagnostic purposes or information access
+        </Headline>
+
+        <RelevanceField value={config.relevant} onChange={updateConfig('relevant')} />
+
+        <Field id="tree-explanation" label="Skill explanation"
+               value={config.explanation} onChange={updateConfig('explanation')} />
+        <Field id="tree-clarification" label="Skill clarification"
+               value={config.clarification} onChange={updateConfig('clarification')} />
+        <Field id="tree-clarification" label="Valid keywords (comma separated)"
+          value={config.keywords} onChange={updateConfig('keywords')} />
+        <DecisionTreeComponent tree={config.tree} onChange={updateConfig('tree')} />
+      </div>
+    )
+  }
+}
+
+class DecisionTreeComponent extends Component {
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      path: this._buildDefaultPath(props.tree.nodes, props.tree.initial),
+      editingOptionTo: null
```

Comment:
Add a comment to what this represent/data that hold. I think it's not obvious.

For how is used maybe something like `triggerFocusOnOptionToNode` would be more accurate